sklearn/neural_network/multilayer_perceptron.py

```diff
@@ -908,6 +908,12 @@ def _validate_input(self, X, y, incremental):
             self._label_binarizer = LabelBinarizer()
             self._label_binarizer.fit(y)
             self.classes_ = self._label_binarizer.classes_
+        elif self.warm_start:
+            classes = unique_labels(y)
+            if set(classes) != set(self.classes_):
+                raise ValueError("`y` has classes not in `self.classes_`."
```

Comment:
Let's make this clearer: "warm_start can only be used where the new `y` has the same classes as in the previous call to fit. Previously, got %r, now %r."